backend/portal/api/portal_api.py

```diff
@@ -434,6 +425,72 @@ def get_collection_index():
     return make_response(jsonify(response), 200)
 
 
+def get_user_private_collections(token_info: Optional[dict] = None):
+    user_info = UserInfo(token_info)
+
+    # Get all private collections the user has write access to
+    if user_info.is_none():
+        all_user_private_collections = []
+    elif user_info.is_super_curator():
+        all_user_private_collections = get_business_logic().get_collections(CollectionQueryFilter(is_published=False))
+    else:
+        all_user_private_collections = get_business_logic().get_collections(
+            CollectionQueryFilter(is_published=False, owner=user_info.user_id)
+        )
+    return all_user_private_collections
+
+
+def get_user_collection_index(token_info):
+    """
+    Returns a list of collections that are published and active.
+    Also returns a subset of fields and not datasets.
+    """
+
+    user_info = UserInfo(token_info)
+
+    # Get all user collections (public and private)
+    public_collections = get_business_logic().get_collections(CollectionQueryFilter(is_published=True))
+    user_private_collections = get_user_private_collections(token_info)
+
+    response = []
+    for collection in itertools.chain(public_collections, user_private_collections):
+        transformed_collection = {
+            "access_type": "WRITE" if user_info.is_user_owner_or_allowed(collection.owner) else "READ",
+            "created_at": collection.created_at,
+            "curator_name": collection.curator_name,
+            "id": collection.collection_id.id,
+            "name": collection.metadata.name,
+            "owner": collection.owner,
+            "visibility": "PRIVATE" if collection.published_at is None else "PUBLIC",
+        }
+
+        if collection.publisher_metadata is not None:
+            transformed_collection["publisher_metadata"] = _publisher_metadata_to_response(
+                collection.publisher_metadata
+            )
+
+        if collection.is_unpublished_version():
+            transformed_collection["id"] = collection.version_id.id
+        else:
+            transformed_collection["id"] = collection.collection_id.id
+
+        if not collection.is_published():
+            transformed_collection["revision_of"] = collection.collection_id.id
+        else:
+            transformed_collection["revision_of"] = None
+
+        if collection.is_published():
+            transformed_collection["published_at"] = collection.canonical_collection.originally_published_at
+            transformed_collection["revised_at"] = collection.published_at
```

Comment:
if this Collection has only been published one time, then this value should probably be `None`. Though I don't know what the FE is expecting... if the FE accepts such a value, then it should be `None`.

Reply:
Let's make sure the tests cover this code path

Reply:
revised_at == published_at is the standard way to indicate that a collection has never been revised. This line exists (once ) elsewhere so so I think this is ok.